### PR TITLE
feat: [C4R-154] added dto validation

### PR DESCRIPTION
**Pull request by OlteanuCosmin**
- added dto validation for every dto so far
- switched from manual swagger decorators to swagger plugin to reduce boilerplate

### CODE REVIEW

File: backend/package.json [hunk 1]
```diff
@@ -55,7 +55,8 @@
     "rimraf": "^3.0.2",
     "rxjs": "^7.2.0",
     "swagger-ui-express": "^4.4.0",
-    "typeorm": "^0.3.6"
+    "typeorm": "^0.3.6",
+    "validator": "^13.7.0"
```
Comment (radulescuandrew) [inline backend/package.json:59]: La ce e folosit asta?

**Review by radulescuandrew — CHANGES_REQUESTED** [review]

File: backend/package.json (hunk 1 above)
Comment (OlteanuCosmin) [reply backend/package.json:59]: Am vrut sa-l folosesc dar nu am mai avut nevoie, si am uitat sa-l dezinstalez inainte sa dau push.

File: backend/src/modules/organization/dto/report.dto.ts [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
 export class Report {
-  @ApiProperty()
+  @IsString()
+  @Matches(/^(http(s)?:\/\/)/)
   link: string;
 
-  @ApiProperty()
+  @IsNumber()
+  @Min(1900)
+  @Max(2022)
```
[the remark this answers is not in the document]
Comment (OlteanuCosmin) [reply backend/src/modules/organization/dto/report.dto.ts]: Am modificat acum sa ia valoarea maxima current year.

**Review by radulescuandrew — APPROVED** [review]

File: backend/src/modules/organization/dto/create-contact.dto.ts [hunk 3]
```diff
@@ -1,27 +1,24 @@
...
+  Matches,
 } from 'class-validator';
 
 export class CreateContactDto {
-  @ApiProperty()
   @IsString()
   @IsNotEmpty()
-  @MaxLength(100)
+  @Length(10, 100)
+  @Matches(/^[a-zA-Z-]*$/)
```
Comment (birloiflorian) [inline backend/src/modules/organization/dto/create-contact.dto.ts:14]: FullName should support space and '-'.

File: backend/src/modules/organization/dto/create-contact.dto.ts [hunk 4]
```diff
@@ -1,27 +1,24 @@
...
   fullName: string;
 
-  @ApiProperty()
   @IsString()
   @IsNotEmpty()
-  @MaxLength(15)
-  @MinLength(10)
+  @Length(10, 10)
   phone: string;
```
Comment (birloiflorian) [inline backend/src/modules/organization/dto/create-contact.dto.ts:20]: Phone should support only numeric characters

File: backend/src/modules/organization/dto/create-organization-activity.dto.ts [hunk 5]
```diff
@@ -1,64 +1,54 @@
...
 import {
   IsArray,
   IsBoolean,
   IsNotEmpty,
   IsNumber,
   IsOptional,
   IsString,
+  Length,
+  Matches,
   MaxLength,
```
Comment (birloiflorian) [inline backend/src/modules/organization/dto/create-organization-activity.dto.ts:10]: Remove unused imports

**Review by birloiflorian — CHANGES_REQUESTED** [review]

**Review by birloiflorian — APPROVED** [review]

**Pull request merged**